freqtrade/data/dataprovider.py

```diff
@@ -123,12 +89,82 @@ def get_analyzed_dataframe(self, pair: str, timeframe: str) -> Tuple[DataFrame,
 
             return (DataFrame(), datetime.fromtimestamp(0, tz=timezone.utc))
 
+    @property
+    def runmode(self) -> RunMode:
+        """
+        Get runmode of the bot
+        can be "live", "dry-run", "backtest", "edgecli", "hyperopt" or "other".
+        """
+        return RunMode(self._config.get('runmode', RunMode.OTHER))
+
+    def current_whitelist(self) -> List[str]:
+        """
+        fetch latest available whitelist.
+
+        Useful when you have a large whitelist and need to call each pair as an informative pair.
+        As available pairs does not show whitelist until after informative pairs have been cached.
+        :return: list of pairs in whitelist
+        """
+
+        if self._pairlists:
+            return self._pairlists.whitelist.copy()
+        else:
+            raise OperationalException("Dataprovider was not initialized with a pairlist provider.")
+
+    def clear_cache(self):
+        """
+        Clear pair dataframe cache.
+        """
+        self.__cached_pairs = {}
+
+    # Exchange functions
+
+    def refresh(self,
+                pairlist: ListPairsWithTimeframes,
+                helping_pairs: ListPairsWithTimeframes = None) -> None:
+        """
+        Refresh data, called with each cycle
+        """
+        if self._exchange is None:
+            raise OperationalException('Exchange is not available to DataProvider.')
+        if helping_pairs:
+            self._exchange.refresh_latest_ohlcv(pairlist + helping_pairs)
+        else:
+            self._exchange.refresh_latest_ohlcv(pairlist)
+
+    @property
+    def available_pairs(self) -> ListPairsWithTimeframes:
+        """
+        Return a list of tuples containing (pair, timeframe) for which data is currently cached.
+        Should be whitelist + open trades.
+        """
+        if self._exchange is None:
+            raise OperationalException('Exchange is not available to DataProvider.')
```

Comment:
i'm ... not sure if this is correct behaviour.

I don't think too many will use this method - but i think this could return the list of pairs in `__cached_pairs` in this case (maybe even in every case ...)